tmt/utils/__init__.py

```diff
@@ -6661,3 +6664,99 @@ def render_rst(text: str, logger: Logger) -> str:
     document.walkabout(visitor)
 
     return visitor.rendered
+
+
+def import_jira() -> None:
+    """ Import Python Jira library """
+    try:
+        global JIRA
+        from jira import JIRA
+    except ImportError:
+        raise GeneralError(
+            "Install 'tmt+link-jira' to use the Jira linking.")
+
+
+def jira_link(
+        nodes: Sequence[Union['tmt.base.Test', 'tmt.base.Plan', 'tmt.base.Story']],
+        links: 'tmt.base.Links',
+        logger: tmt.log.Logger,
+        separate: bool = False) -> None:
+    """ Link the object to Jira issue and create the URL to tmt web service """
+    import_jira()
+
+    def create_url_params(tmt_object: tmt.base.Core) -> dict[str, Any]:
+        tmt_type = tmt_object.__class__.__name__.lower()
+        fmf_id = tmt_object.fmf_id
+
+        url_params: dict[str, Any] = {
+            'format': 'html',
+            f'{tmt_type}-url': fmf_id.url,
+            f'{tmt_type}-name': fmf_id.name
+            }
+
+        if fmf_id.path and isinstance(fmf_id.git_root, Path):
+            fmf_path = fmf_id.path.relative_to(fmf_id.git_root).resolve().as_posix()
+            fmf_path = fmf_path.removeprefix(fmf_id.git_root.as_posix())
+            url_params[f'{tmt_type}-path'] = fmf_path.lstrip('/')
+
+        if fmf_id.ref:
+            url_params[f'{tmt_type}-ref'] = fmf_id.ref
+
+        return url_params
+
+    def create_url(baseurl: str, url_params: dict[str, str]) -> str:
+        return urllib.parse.urljoin(baseurl, '?' + urllib.parse.urlencode(url_params))
+
+    logger.print(
+        f'Linking {fmf.utils.listed([type(node).__name__.lower() for node in nodes])}'
+        f' to Jira issue.')
+
+    # Setup config tree
+    config_tree = tmt.utils.Config()
+    # Linking is not setup in config, therefore user does not want to use linking
+    linking_node = config_tree.fmf_tree.find('/user/linking')
```

Comment:
I am quite sure that we agreed on `linking`, can someone else confirm?

Reply:
If I remember correctly, there was this idea to pick config section names which would be aligned with the commands related to them. So here, for `tmt link` it would by `link`. In a similar way, could/should have a `try` section for `tmt try`. Would that make sense?

Reply:
Fixed in 8f8cc015.